### PR TITLE
use suppressions

### PR DESCRIPTION
**Pull request by cataggar**
This moves to use the `suppressions` configuration and removes the deprecated `suppress` directives.

Trying out https://github.com/Azure/autorest/issues/4620 .

```
PS C:\Users\cataggar\io\azure-rest-api-specs> autorest --v3 --version:3.9.2 --spectral --validation --azure-validator --semantic-validator=false --model-validator=false --openapi-type=arm --use=@microsoft.azure/openapi-validator@2.0.0-beta.3 --tag=package-2022-05-01 specification\vmware\resource-manager\readme.md > out.txt
Unexpected error while logging TypeError: text.replace is not a function
    at color (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12498:10)
    at PrettyLogFormatter.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:11819:54)
    at ConsoleLoggerSink.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:11573:37)
    at ConsoleLogger.emitLog (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12124:18)
    at ConsoleLogger.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12120:14)
    at AutorestAsyncLogger.emitLog (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12157:18)
    at AutorestAsyncLogger.logMessageAsync (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12171:14)
    at C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12215:17
```

### CODE REVIEW

**Comment by cataggar** [comment]
[out.txt](https://github.com/cataggar/azure-rest-api-specs/files/9676391/out.txt)

**Comment by cataggar** [comment]
[out.txt](https://github.com/cataggar/azure-rest-api-specs/files/9676391/out.txt)

File: specification/vmware/resource-manager/readme.md [hunk 1]
```diff
@@ -291,7 +294,10 @@ directive:
       - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/dnsServices/{dnsServiceId}"].delete
       - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/dnsZones/{dnsZoneId}"].delete
       - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/publicIPs/{publicIPId}"].delete
-    reason: It is a breaking change to update the parameters order.
+    
+  - code PathResourceProviderNamePascalCase
```
Comment (cataggar) [inline specification/vmware/resource-manager/readme.md]: Addind the missing `:` did not help.

**Comment by timotheeguerin** [comment]
is the `text.replace` error new from upgrading to 3.9.2?

File: specification/vmware/resource-manager/readme.md [hunk 2]
```diff
@@ -269,8 +269,10 @@ swagger-to-sdk:
 
 ## Suppression
 ```
```
Comment (timotheeguerin) [inline specification/vmware/resource-manager/readme.md]:
````suggestion
```yaml
````
missing the yaml tag here

**Comment by cataggar** [comment]
> is the text.replace error new from upgrading to 3.9.2?

@timotheeguerin, yes, it does not happen with `next` which is `3.9.0-dev.3` https://www.npmjs.com/package/@autorest/core .

**Comment by timotheeguerin** [comment]
> > is the text.replace error new from upgrading to 3.9.2?
> 
> @timotheeguerin, yes, it does not happen with `next` which is `3.9.0-dev.3` https://www.npmjs.com/package/@autorest/core .

That error seems to be caused by something logging an undefined log message, guessing one of the extensions must be doing that.

I can't seem to reproduce this error though, could you try to dig into this and see (removing different flags in the cli above) what might be causing this

**Comment by cataggar** [comment]
I added `--debug` if that helps.

```
PS C:\Users\cataggar\io\azure-rest-api-specs> autorest --v3 --version:3.9.2 --spectral --validation --azure-validator --semantic-validator=false --model-validator=false --openapi-type=arm --use=@microsoft.azure/openapi-validator@2.0.0-beta.3 --tag=package-2022-05-01 specification\vmware\resource-manager\readme.md --debug
AutoRest code generation utility [cli version: 3.6.2; node: v16.14.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: 3.9.2.
debug   | [1.25 s] Network Enabled: true
debug   | [2.35 s] The most recent installed version is 3.9.2
debug   | [2.36 s] Starting @autorest/core from C:\Users\cataggar\.autorest\@autorestcore@3.9.2\nodemodules\@autorest\core
info    |    Loading AutoRest core      'C:\Users\cataggar\.autorest\@autorestcore@3.9.2\nodemodules\@autorest\core\dist' (3.9.2)
verbose | [1.28 s]    Initial configuration file 'file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/readme.md'
verbose | [1.28 s]    Including configuration file 'file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/readme.md'
verbose | [1.34 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/default-configuration.md'
verbose | [1.36 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/inspect.md'
verbose | [1.38 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/directives.md'
verbose | [1.41 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/pipeline.md'
verbose | [1.43 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/loader-openapi.md'
verbose | [1.45 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/miscellaneous.md'
verbose | [1.47 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-powershell.md'
verbose | [1.49 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-azureresourceschema.md'
verbose | [1.50 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-csharp.md'
verbose | [1.52 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-go.md'
verbose | [1.54 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-java.md'
verbose | [1.55 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-nodejs.md'
verbose | [1.57 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-php.md'
verbose | [1.59 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-python.md'
verbose | [1.61 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-ruby.md'
verbose | [1.63 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-typescript.md'
verbose | [1.65 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-validators.md'
verbose | [1.67 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-az.md'
verbose | [1.68 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-terraform.md'
verbose | [1.70 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/plugin-azure-functions.md'
verbose | [1.72 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/graphs.md'
verbose | [1.74 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@autorestcore@3.9.2/nodemodules/@autorest/core/dist/resources/help-configuration.md'
info    |    Loading AutoRest extension '@microsoft.azure/openapi-validator' (2.0.0-beta.3->2.0.0-beta.3)
verbose | [1.89 s]    Including extension configuration file 'file:///C:/Users/cataggar/.autorest/@microsoft.azureopenapi-validator@2.0.0-beta.3/nodemodules/@microsoft.azure/openapi-validator/readme.md'
verbose | [1.89 s]    Including configuration file 'file:///C:/Users/cataggar/.autorest/@microsoft.azureopenapi-validator@2.0.0-beta.3/nodemodules/@microsoft.azure/openapi-validator/readme.md'
debug   | [8.87 s] swagger-document/loader-swagger - START inputs = 0
debug   | [8.87 s] openapi-document/loader-openapi - START inputs = 0
debug   | [8.87 s] pipeline-emitter - START inputs = 0
debug   | [8.87 s] configuration-emitter - START inputs = 0
debug   | [8.87 s] pipeline-emitter - END [0.01 s][67 MB]
debug   | [8.87 s] configuration-emitter - END [0.01 s][67 MB]
verbose | [8.91 s] Reading OpenAPI 2.0 file file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json
debug   | [8.97 s] openapi-document/loader-openapi - END [0.05 s][73 MB]
verbose | [8.97 s] Reading $ref'd file file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/common-types/resource-management/v1/types.json
debug   | [8.97 s] openapi-document/individual/transform - SKIPPING
debug   | [8.97 s] openapi-document/individual/schema-validator-openapi - SKIPPING
debug   | [8.97 s] openapi-document/individual/full-ref-resolver - SKIPPING
debug   | [8.97 s] openapi-document/individual/identity - SKIPPING
verbose | [8.98 s] Reading $ref'd file file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/common-types/resource-management/v3/types.json
debug   | [8.99 s] swagger-document/loader-swagger - END [0.13 s][63 MB]
debug   | [8.99 s] swagger-document/individual/transform - START inputs = 3
debug   | [8.99 s] swagger-document/openapi-validator - START inputs = 3
debug   | [8.99 s] swagger-document/spectral - START inputs = 3
debug   | [8.99 s] swagger-document/individual/transform - END [0 s][63 MB]
debug   | [8.99 s] swagger-document/individual/schema-validator-swagger - START inputs = 3
debug   | [9.21 s] swagger-document/individual/schema-validator-swagger - END [0.22 s][71 MB]
debug   | [9.21 s] swagger-document/individual/full-ref-resolver - START inputs = 3
verbose | [9.29 s] Validating 'file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json'
verbose | [9.30 s] Validating 'file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json'
debug   | [9.48 s] swagger-document/individual/full-ref-resolver - END [0.26 s][59 MB]
debug   | [9.48 s] swagger-document/identity - START inputs = 3
debug   | [9.48 s] swagger-document/individual/identity - START inputs = 3
debug   | [9.48 s] swagger-document/identity - END [0 s][59 MB]
debug   | [9.48 s] swagger-document/individual/identity - END [0 s][59 MB]
debug   | [9.48 s] openapi-document/openapi-document-converter - START inputs = 3
debug   | [9.62 s] openapi-document/openapi-document-converter - END [0.14 s][70 MB]
debug   | [9.63 s] openapi-document/transform - START inputs = 3
debug   | [9.64 s] openapi-document/transform - END [0 s][76 MB]
debug   | [9.64 s] openapi-document/semantic-validator - START inputs = 3
debug   | [9.66 s] openapi-document/semantic-validator - END [0.02 s][83 MB]
debug   | [9.67 s] openapi-document/allof-cleaner - START inputs = 3
debug   | [9.69 s] openapi-document/allof-cleaner - END [0.02 s][72 MB]
debug   | [9.69 s] openapi-document/tree-shaker - START inputs = 3
debug   | [9.89 s] openapi-document/tree-shaker - END [0.2 s][92 MB]
debug   | [9.90 s] openapi-document/multi-api-merger - START inputs = 3
debug   | [10.07 s] openapi-document/multi-api-merger - END [0.16 s][103 MB]
debug   | [10.07 s] openapi-document/components-cleaner - START inputs = 1
debug   | [10.19 s] openapi-document/components-cleaner - END [0.12 s][102 MB]
debug   | [10.19 s] openapi-document/component-modifiers - START inputs = 1
debug   | [10.21 s] openapi-document/component-modifiers - END [0.01 s][101 MB]
debug   | [10.21 s] openapi-document/api-version-parameter-handler - START inputs = 1
debug   | [10.21 s] openapi-document/api-version-parameter-handler - END [0 s][101 MB]
debug   | [10.21 s] openapi-document/profile-filter - START inputs = 1
debug   | [10.21 s] openapi-document/profile-filter - END [0 s][101 MB]
debug   | [10.21 s] openapi-document/model-deduplicator - START inputs = 1
debug   | [10.88 s] swagger-document/spectral - END [1.89 s][113 MB]
debug   | [11.03 s] openapi-document/model-deduplicator - END [0.81 s][109 MB]
debug   | [11.03 s] openapi-document/compose - START inputs = 1
debug   | [11.18 s] openapi-document/compose - END [0.14 s][128 MB]
debug   | [11.18 s] openapi-document/identity - START inputs = 1
debug   | [11.18 s] openapi-document/identity - END [0 s][119 MB]
debug   | [11.18 s] openapi-document/emitter - START inputs = 1
debug   | [11.18 s] openapi-document/emitter - END [0 s][119 MB]
Unexpected error while logging TypeError: text.replace is not a function
    at color (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12498:10)
    at PrettyLogFormatter.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:11819:54)
    at ConsoleLoggerSink.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:11573:37)
    at ConsoleLogger.emitLog (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12124:18)
    at ConsoleLogger.log (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12120:14)
    at AutorestAsyncLogger.emitLog (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12157:18)
    at AutorestAsyncLogger.logMessageAsync (C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12171:14)
    at C:\Users\cataggar\.autorest\@autorest_core@3.9.2\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:12215:17
debug   | [13.91 s] swagger-document/openapi-validator - END [4.92 s][127 MB]
debug   | [13.91 s] Clearing Folders.
debug   | [13.91 s] Writing Outputs.
info    | Autorest completed in 13.91s. 0 files generated.
debug   | [13.91 s] Shutting Down.
debug   | [13.91 s] Exiting.
```

**Comment by cataggar** [comment]
@timotheeguerin, the error went away when I suppressed `ArmResourcePropertiesBag` in https://github.com/Azure/azure-rest-api-specs/pull/20873/commits/99ba0087323c237db0072bdac142231909cec4eb. In json, the error was:

> {"pluginName":"spectral","extensionName":"@microsoft.azure/openapi-validator","level":"error","message":{"message":"Top level property names should not be repeated inside the properties bag for ARM resource '{0}'. Properties [properties.type] conflict with ARM top level properties. Please rename these."},"code":"ArmResourcePropertiesBag","details":{"jsonpath":["definitions","PlacementPolicy"],"validationCategory":"ARMViolation","providerNamespace":false,"resourceType":false,"range":{"start":{"line":7806,"column":24},"end":{"line":7821,"column":6}}},"source":[{"document":"file:///C:/Users/cataggar/io/azure-rest-api-specs/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json","position":{"line":7806,"column":5}}]}

**Pull request merged**